chalice/package.py

```diff
@@ -357,15 +386,255 @@ def _register_cfn_resource_name(self, name):
+                resource.resource_name] = {
+                    'rule': '${aws_cloudwatch_event_rule.%s.name}' % (
+                        resource.resource_name),
+                    'target_id': resource.resource_name,
+                    'arn': self._fref(resource.lambda_function)
+        }
+        template['resource'].setdefault(
+            'aws_lambda_permission', {})[
+                resource.resource_name] = {
+                    'function_name': self._fref(resource.lambda_function),
+                    'action': 'lambda:InvokeFunction',
+                    'principal': 'events.amazonaws.com',
+                    'source_arn': "${aws_cloudwatch_event_rule.%s.arn}" % (
+                        resource.resource_name)
+        }
+
+    def _generate_lambdafunction(self, resource, template):
+        # type: (models.LambdaFunction, Dict[str, Any]) -> None
+
+        func_definition = {
+            'function_name': resource.function_name,
+            'runtime': resource.runtime,
+            'handler': resource.handler,
+            'memory_size': resource.memory_size,
+            'tags': resource.tags,
+            'timeout': resource.timeout,
+            'source_code_hash': '${filebase64sha256("%s")}' % (
+                resource.deployment_package.filename),
+            'filename': resource.deployment_package.filename}
+
+        if resource.security_group_ids and resource.subnet_ids:
+            func_definition['vpc_config'] = {
+                'subnet_ids': resource.subnet_ids,
+                'security_group_ids': resource.security_group_ids
+            }
+        if resource.reserved_concurrency is not None:
+            func_definition['reserved_concurrent_executions'] = (
+                resource.reserved_concurrency
+            )
+        if resource.environment_variables:
+            func_definition['environment'] = {
+                'variables': resource.environment_variables
+            }
+        if resource.layers:
+            func_definition['layers'] = list(resource.layers)
+
+        if isinstance(resource.role, models.ManagedIAMRole):
+            func_definition['role'] = '${aws_iam_role.%s.arn}' % (
+                resource.role.resource_name)
+        else:
+            # resource is a PreCreatedIAMRole.
+            role = cast(models.PreCreatedIAMRole, resource.role)
+            func_definition['role'] = role.role_arn
+
+        template['resource'].setdefault('aws_lambda_function', {})[
+            resource.resource_name] = func_definition
+
+    def _generate_restapi(self, resource, template):
+        # type: (models.RestAPI, Dict[str, Any]) -> None
+
+        # typechecker happiness
+        swagger_doc = cast(Dict, resource.swagger_doc)
+        template['resource'].setdefault('aws_api_gateway_rest_api', {})[
+            resource.resource_name] = {
+                'body': json.dumps(swagger_doc),
+                # Terraform will diff explicitly configured attributes
```

Comment:
That's unfortunate. We'll probably have to set these for minimum compression, endpoint type, and policy as well.

Reply:
yes, all my half-dozen pull requests conflict based on the way the project is structured and will need updates in response to any trunk changes. thats part of the reason contributor pull requests should be prioritized instead of allowed to languish imo. i'll update to this supports whats currently merged, else its a responsibility of feature contributors and maintainers to ensure parity.